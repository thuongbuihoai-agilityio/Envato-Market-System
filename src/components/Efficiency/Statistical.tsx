import { Box, Flex, Text } from '@chakra-ui/react';
import { memo } from 'react';

type TStatisticalProps = {
  title: string;
  color: string;
  value: number;
};

const StatisticalComponent = ({
  title,
  color,
  value,
}: TStatisticalProps): JSX.Element => (
  <Flex dir="row" w="full" alignItems="center">
    <Box bg={color} w="10px" height="10px" rounded="50%" />
    <Text ml={3} variant="textSm">
      {title}
    </Text>
    <Text
      fontSize="sm"
      fontWeight="bold"
      color="text.primary"
      sx={{ marginLeft: 'auto' }}
    >
      {value}%
    </Text>
  </Flex>
);

const Statistical = memo(StatisticalComponent);

export default Statistical;
