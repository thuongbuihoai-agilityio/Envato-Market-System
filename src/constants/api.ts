export const USERS_API = 'https://65660396eb8bb4b70ef2c750.mockapi.io';

// TODO: Will provide latter
export const BASE_API = '';
export const STATISTICAL_API = 'https://succinct-shrouded-acorn.glitch.me';

export const END_POINTS = {
  AUTHEN_USERS: '/authenUsers',
  USERS: '/users',
  TRANSACTIONS: '/transactions',
  STATISTICS: '/statistics',
};

export const SEARCH_PARAM = {
  EMAIL: 'email',
  PASSWORD: 'password',
};
