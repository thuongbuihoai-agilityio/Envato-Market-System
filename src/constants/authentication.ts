export const SUPER_ADMIN = 'Super Admin';

export const MEMBER = 'member';
