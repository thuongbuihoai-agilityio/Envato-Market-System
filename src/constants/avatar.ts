export enum AVATAR_POSITION {
  BEFORE = 'before',
  AFTER = 'after',
}
