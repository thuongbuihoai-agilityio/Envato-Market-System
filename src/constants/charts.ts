export const STROKE_COLORS = ['#22C55E', '#EAB308', '#EDF2F7', '#23262B'];
export const REVENUE_FLOW_COLORS = ['#EAB308', '#22C55E', '#FF784B', '#EDF2F7'];

export const MONTHS = [
  'Jan',
  'Feb',
  'Mar',
  'April',
  'May',
  'Jun',
  'July',
  'Aug',
  'Sep',
  'Oct',
  'Nov',
  'Dec',
];

export const REVENUE_FLOW_STATUS = ['Pending', 'Signed', 'Lost'];
