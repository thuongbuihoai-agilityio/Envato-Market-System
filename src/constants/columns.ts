// Types
import { TDataSource } from '@app/interfaces';

export const COLUMNS_DASHBOARD = (
  onRenderHead: (title: string, key: string) => void,
  onRenderBody: ({ id, image, name }: TDataSource) => void,
  onRenderActionIcon: (data: TDataSource) => void,
) => [
  {
    title: 'Customer name',
    key: 'name',
    renderHead: onRenderHead,
    renderBody: onRenderBody,
  },
  {
    title: 'Email',
    key: 'email',
    renderHead: onRenderHead,
  },
  {
    title: 'Location',
    key: 'location',
    renderHead: onRenderHead,
  },
  {
    title: 'Spent',
    key: 'spent',
    renderHead: onRenderHead,
  },
  {
    title: '',
    key: 'action',
    renderBody: onRenderActionIcon,
    renderHead: onRenderHead,
  },
];

export const COLUMNS_HISTORY = (
  renderHead: (title: string, key: string) => void,
  renderBody: ({ id, image, name }: TDataSource) => void,
  renderPaymentStatus: ({ paymentStatus }: TDataSource) => void,
  renderTransactionStatus: ({ paymentStatus }: TDataSource) => void,
  renderActionIcon: (data: TDataSource) => void,
) => [
  {
    title: 'Customer name',
    key: 'name',
    renderHead: renderHead,
    renderBody: renderBody,
  },
  {
    title: 'Date',
    key: 'date',
    renderHead: renderHead,
  },
  {
    title: 'Amount',
    key: 'spent',
    renderHead: renderHead,
  },
  {
    title: 'Payment',
    key: 'paymentStatus',
    renderHead: renderHead,
    renderBody: renderPaymentStatus,
  },
  {
    title: 'Status',
    key: 'transactionStatus',
    renderHead: renderHead,
    renderBody: renderTransactionStatus,
  },
  {
    title: '',
    key: 'action',
    renderBody: renderActionIcon,
  },
];
