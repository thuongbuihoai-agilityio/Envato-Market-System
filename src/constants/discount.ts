export const DEFAULT_DISCOUNT_PERCENTAGE = 60;
export const LIMIT_OF_BONUS = 5;
