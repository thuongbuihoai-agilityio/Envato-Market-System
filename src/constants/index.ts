export * from './routers';
export * from './time';
export * from './images';
export * from './form';
export * from './messages';
export * from './regex';
export * from './texts';
export * from './themes';
export * from './api';
export * from './images';
export * from './menu';
export * from './sidebar';
export * from './charts';
export * from './sidebar';
export * from './common';
export * from './options';
export * from './transaction';
export * from './store';
export * from './status';
export * from './pagination';
export * from './columns';
