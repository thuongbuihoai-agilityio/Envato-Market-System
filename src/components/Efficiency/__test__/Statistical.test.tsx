import { render } from '@testing-library/react';

// Components
import Statistical from '../Statistical';
import { STROKE_COLORS } from '@app/constants';

describe('EfficiencyInfo component', () => {
  it('should render correctly', () => {
    const { container } = render(
      <Statistical color={STROKE_COLORS[0]} title="Spending" value={1234} />,
    );

    expect(container).toMatchSnapshot();
  });
});
