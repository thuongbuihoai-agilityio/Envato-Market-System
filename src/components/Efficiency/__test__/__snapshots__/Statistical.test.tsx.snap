// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Statistical component Should render correctly 1`] = `
<div>
  <div
    class="css-n2i4qd"
    dir="row"
  >
    <div
      class="css-1tzz1x1"
    />
    <p
      class="chakra-text css-9n3bbc"
    >
      Spending
    </p>
    <p
      class="chakra-text css-15h4n6s"
    >
      1234
      %
    </p>
  </div>
</div>
`;
